nntrainer/optimizers/optimizer_wrapped.cpp

```diff
@@ -0,0 +1,144 @@
+// SPDX-License-Identifier: Apache-2.0
+/**
+ * Copyright (C) 2021 Parichay Kapoor <pk.kapoor@samsung.com>
+ *
+ * @file   optimizer_wrapped.cpp
+ * @date   10 December 2021
+ * @brief  This is Optimizer Wrapped interface class
+ * @see    https://github.com/nnstreamer/nntrainer
+ * @author Parichay Kapoor <pk.kapoor@samsung.com>
+ * @bug    No known bugs except for NYI items
+ *
+ * @details wraps the optimizer and learning rate scheduler together
+ */
+
+#include <app_context.h>
+#include <common_properties.h>
+#include <lr_scheduler_constant.h>
+#include <lr_scheduler_exponential.h>
+#include <node_exporter.h>
+#include <optimizer_wrapped.h>
+
+namespace nntrainer {
+
+/**
+ * @brief Optimizer wrapped creator with constructor for optimizer
+ */
+std::unique_ptr<OptimizerWrapped>
+createOptimizerWrapped(const ml::train::OptimizerType &type,
+                       const std::vector<std::string> &properties) {
+  auto &ac = nntrainer::AppContext::Global();
+  return createOptimizerWrapped(ac.createObject<OptimizerCore>(type),
+                                properties);
+}
+
+/**
+ * @brief Optimizer wrapped creator with constructor for optimizer
+ */
+std::unique_ptr<OptimizerWrapped>
+createOptimizerWrapped(const std::string &type,
+                       const std::vector<std::string> &properties) {
+  auto &ac = nntrainer::AppContext::Global();
+  return createOptimizerWrapped(ac.createObject<OptimizerCore>(type),
+                                properties);
+}
+
+/**
+ * @brief Optimizer wrapped creator with constructor for optimizer
+ */
+std::unique_ptr<OptimizerWrapped>
+createOptimizerWrapped(std::unique_ptr<OptimizerCore> &&opt,
+                       const std::vector<std::string> &properties) {
+  auto opt_wrapped = std::make_unique<OptimizerWrapped>(std::move(opt));
+
+  opt_wrapped->setProperty(properties);
+  return opt_wrapped;
+}
+
+OptimizerWrapped::OptimizerWrapped(std::unique_ptr<OptimizerCore> &&opt) :
+  optimizer(std::move(opt)),
+  lr_sched(),
+  props(props::LearningRate(), props::DecayRate(), props::DecaySteps()) {}
+
+const std::string OptimizerWrapped::getType() const {
+  return optimizer->getType();
+}
+
+void OptimizerWrapped::setProperty(const std::vector<std::string> &values) {
+  auto remain_props = loadProperties(values, props);
+  optimizer->setProperty(remain_props);
+}
+
+double OptimizerWrapped::getLearningRate(size_t iteration) {
+  return lr_sched->getLearningRate(iteration);
+}
+
+void OptimizerWrapped::applyGradient(RunOptimizerContext &context) {
+  // optimizer->applyGradient(context);
```

Comment:
comment intended?

Reply:
yes, this will be enabled once the optimizer is set in upcoming PR.